cpp/src/arrow/extension/CMakeLists.txt

```diff
@@ -15,10 +15,19 @@
 # specific language governing permissions and limitations
 # under the License.
 
-add_arrow_test(test
-               SOURCES
-               fixed_shape_tensor_test.cc
-               PREFIX
-               "arrow-fixed-shape-tensor")
+if(ARROW_JSON)
+  add_arrow_test(test
+                 SOURCES
+                 fixed_shape_tensor_test.cc
+                 json_test.cc
+                 PREFIX
+                 "arrow-canonical-extensions")
+else()
+  add_arrow_test(test
+                 SOURCES
+                 json_test.cc
```

Comment:
Should this `fixed_shape_tensor_test.cc`?

Reply:
`fixed_shape_tensor` requires rapidjson. I'm actually not sure if `ARROW_JSON` controls if rapidjson gets linked or not, will test.

Reply:
It does. Changed to be more ideomatic.